src/Uno.Extensions.Navigation/RouteExtensions.cs

```diff
@@ -371,7 +371,8 @@ public static IDictionary<string, object> Combine(this IDictionary<string, objec
 			return route;
 		}
 
-		var deepestChild = childRoutes.ToArray().OrderByDescending(x => x.Item2?.ToString().Length ?? 0).FirstOrDefault();
+		//var deepestChild = childRoutes.ToArray().OrderByDescending(x => x.Item2?.ToString().Length ?? 0).FirstOrDefault();
+		var deepestChild = childRoutes.Where(x => x.Item2 is not null).FirstOrDefault();
```

Comment:
Would someone be able to explain why that logic is like this?

Getting the route based on the name length causes a bug where navigating between Tabs or NavViewItems will get the route with larger name and update the address bar with it. This affects chefs and the Navigation sample on Uno.Samples.


![address-bar-issue3](https://github.com/user-attachments/assets/55dd5b19-f738-4136-ade6-9aaf9dfd4fa8)

To make it work I added a logic to make the navigated page on `PanelVisibilityNavigator` to be always the first of the list alongside with changing the logic of largest route name.

cc @kazo0 @nickrandolph